### PR TITLE
Add landing page

### LINKED ISSUE #48: Create landing page
I think we will want a landing page for logged-out users to see. Currently, we're still showing the React logo and a few PDC links. Soon we will have the option of showing the proposals list (#14) in the root, which would require logging in immediately. I think it's worthwhile to have a brief landing page that would introduce the project and maintain the links we currently have.

### PR DESCRIPTION
**Pull request by reefdog**
This PR adds a landing page with lightwight welcome and guidance for use.

**Logged out:**

logged-out

**Logged in:**

logged-in

Some prerequisite changes that I also made:
- Buttons are now `inline-flex`, so they don't have to occupy 100% of available width.
- It's possible to have panels that don't occupy the entire viewport. (This was always the intent, I just didn't complete the work for it because I didn't have a situation that called for it.)
- Added `EmailLink` and `OffsiteLink` components for responsibly handling those respective link types with minimal boilerplate. See story and documentation for use.
- Added styling rules for `<dl>` lists, which are great for key/value pairs like dictionaries or FAQs. This also introduces a tiny new component, `Dli`, that wraps the `<dt>`/`<dd>` children of `<dl>`s.
- Added a `.subtle` text decorator for small, gray text.

Closes #48

### CODE REVIEW

File: src/index.css
Comment (slifty) [inline src/index.css:94]: why doesn't the html spec just have li as a valid child of dl?

Please open a proposed change to the HTML spec thanks.

Comment (slifty) [reply src/index.css:94]: Oh actually, and please boo me off the stage: would it be worth make a dli component so you can just do `<dl><dli>...` instead of the div class blah blah blah?

You could have that component expect / require content for the dd and the dt.  You'd be able to throw out this comment I think, and you'd also be able to avoid repetition of the class stuff in your dl implementations.

File: src/pages/Landing.tsx
Comment (slifty) [inline src/pages/Landing.tsx:56]: I don't know if we're in "move fast and break things" mode for this but this strikes me as a value that should be configurable / pulled from some kind of configuration file or environment variable.

File: src/pages/Landing.tsx
Comment (slifty) [inline src/pages/Landing.tsx:58]: Same suggestion here regarding configuration / not hardcoding values like this.

Comment (slifty) [reply src/pages/Landing.tsx:58]: (Alternatively, as a middle ground solution it might feel less sore if we made it more generic than Jim's name and email address specifically, e.g. something more like "admin@...")

File: src/pages/Landing.tsx
Comment (slifty) [inline src/pages/Landing.tsx:23]: I picked this line to ask this question: do we care about internationalization at this stage of the project?  I assume the answer is no; sorry if I'm rehashing conversations that have already been had!

**Review by slifty — COMMENTED** [review]
I'm going to actually set this up locally but wanted to leave these comments now for your consideration in the mean time.

File: src/pages/Landing.tsx
Comment (reefdog) [reply src/pages/Landing.tsx:23]: That's a great question, and yes, we do care!

Unfortunately while we're accruing debt here, that ship has sailed (in that the site is going to be in user's hands in a matter of minutes without i18n) so that shouldn't block this particular PR. But it's important and I'm glad you noted it.

File: src/index.css
Comment (reefdog) [reply src/index.css:94]: Oh, I like that idea! I wonder if I can lowercase it though so it doesn't look weird between `<dl>` and `<dd>`…

> You could have that component expect / require content for the dd and the dt.

I'd very much like for it to just use the `children` it was handed, e.g.:

```jsx
<dl>
  <dli>
    <dt />
    <dd />
  </dli>
</dl>
```

Do you know if React will let me dictate which elements are allowed as `children`?

**Review by slifty — APPROVED** [review]
Ran this locally -- I don't have an account yet so can't confirm the actual functionality of post-login, but will defer to you on whether you'd like to block on that!

I left comments above but I don't think any of them are blocking comments unless you agree with them and would like to incorporate them.

File: src/index.css
Comment (reefdog) [reply src/index.css:94]: Implemented. This was a great suggestion, thanks @slifty!

> Do you know if React will let me dictate which elements are allowed as children? 

Looks like no, so this just accepts any children. Ah well, the some holds true for any HTML element: if you pass it something invalid, the browser will scream at you.

File: src/pages/Landing.tsx
Comment (reefdog) [reply src/pages/Landing.tsx:55]: Since this is for a pilot group of like 20 people atm, all of whom know Jim, the clarity of this direction is actually preferred. As a point of process, we may want a more general inbox, but that's a decision out of scope of this implementation so I'm gonna move forward on this point!

I mean, I get that this could be `Email ${ACCOUNT_ADMIN_NAME}` and fair enough! But I'll reserve that for the first time we need to change it and/or have downtime before getting the pilot participant logins out this afternoon.

Comment (reefdog) [reply src/pages/Landing.tsx:55]: @slifty My response here was initially written in response to [this comment](https://github.com/PhilanthropyDataCommons/data-viewer/pull/69#discussion_r1160888777) and then copied here with slight changes, so some of it actually addresses your point there.

**Comment by reefdog** [comment]
@slifty That was great feedback, thanks! You're spot on that we've got values hardcoded that shouldn't be, but I'm happy punting on that for now until we better understand how often we'll be doing that, how we'd want to store those values, how we'd want to pass them into our i18n, etc.

On which point, good call on i18n. We intended to have that from launch, and currently do not, and should! I'll open the issue.

I added a `Dli` component (can't go with `dli` as much as I'd like to) as syntactic sugar for working with `<dl>`s, and slipped in some new components for dealing with email and offsite links more responsibly. Re-review appreciated!

**Review by slifty — APPROVED** [review]
Looks great!  Thanks!

File: src/index.css
Comment (jasonaowen) [inline src/index.css:4]: I found this a little hard to read, and looked up what the [accessibility standards for color contrast](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Color_contrast) are. In short, the minimum is 4.5:1, and ideal is 7:1; [#999 against white](https://webaim.org/resources/contrastchecker/?fcolor=999999&bcolor=FFFFFF) has a ratio of 2.84:1.

`#757575` has a ratio of 4.6:1, which clears the minimum; is that subtle enough?

```suggestion
  --color--gray--medium-dark: #757575;
```

File: src/index.css
Comment (jasonaowen) [inline src/index.css:92]: Can you please say more about this? Why is adding a `div` into the tree better than adding a class on `<dl>`? I don't understand the problem this solves!

**Comment by reefdog** [comment]
Huzzah! 🚀

File: src/components/OffsiteLink.tsx
Comment (jasonaowen) [inline src/components/OffsiteLink.tsx:23]: I'm going to push back a bit on this, too!

Why do we want links to open in a new tab? Users have the ability to choose whether to open in a new window/tab in every modern browser. If they click an external link and decide they wanted it to be in a new tab after all, they can hit the back button and middle click. If we force it to open in a new tab, users can't make that decision.

**Review by jasonaowen — COMMENTED** [review]
Sorry for the late review!

File: src/pages/Landing.tsx
Comment (bickelj) [reply src/pages/Landing.tsx:26]: `s/Philanythropy/Philanthropy/g`

Comment (reefdog) [reply src/pages/Landing.tsx:26]: Oh nooooo

File: src/index.css
Comment (reefdog) [reply src/index.css:4]: Great note, will change!

File: src/index.css
Comment (reefdog) [reply src/index.css:92]: Somewhat infamously, definition lists don't have individual children. Instead, their children are a soup of terms and definitions, e.g.:

```html
<dl>
  <dt>Dog</dt>
  <dd>Furry quadruped with a sweet nature.</dd>
  <dt>Cat</dt>
  <dd>Furry quadruped with a sneaky nature.</dd>
</dl>
```

It's very common to want to style the `dt`/`dd` pairs as a unit, e.g. to put spaces between them so they appear somewhat like:

```
Dog
Furry quadruped with a sweet nature.

Cat
Furry quadruped with a sneaky nature.
```

Of course it's possible to style them by, say, adding margin below `dt` or above `dd`, and using `:first-child` and `:last-child` to get rid of boundary margins.

But the whole thing is very unsemantic, and doesn't let you do neat things like "use flexbox column layout and put a gap between each pair". So it's now valid to wrap them in a `div` — acting equivalent of an `li` — while still getting the semantics of `dl`/`dt`/`dd`.

That's certainly a lot of justification, but do you consider it _good_ justification?

File: src/components/OffsiteLink.tsx
Comment (reefdog) [reply src/components/OffsiteLink.tsx:23]: This is where I appreciate your rethinking my priors.

I philosophically agree with you but have lost this battle so often that I not reflexively default to what I expect the client will want ("offsite links open in new tabs"). I am _ecstatic_ to invert this behavior and default to web-native behavior. Excellent pushback.

File: src/index.css
Comment (jasonaowen) [reply src/index.css:92]: That's great justification, and more importantly, great explanation! Thank you, @reefdog, TIL.

**Pull request merged**
